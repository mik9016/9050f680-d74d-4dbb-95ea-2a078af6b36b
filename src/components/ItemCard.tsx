import {
  Card,
  CardActions,
  CardContent,
  CardMedia,
  IconButton,
  CardHeader,
  Button,
  Avatar,
  Typography,
} from '@mui/material'
import AddCircleIcon from '@mui/icons-material/AddCircle'
import React, { useContext } from 'react'
import LocationOnIcon from '@mui/icons-material/LocationOn'
import { GlobalContext } from '../contexts/Context'

export interface Props {
  bandName: string
  avatar: string
  picture: string
  venue: string
  venueName: string
  start: string
  end: string
}

const ItemCard = ({
  bandName,
  avatar,
  picture,
  venue,
  start,
  venueName,
  end,
  ref,
}: Props & { ref?: React.Ref<HTMLDivElement> }): React.ReactElement => {
  const globalCtx = useContext(GlobalContext)

  const infos = {
    bandName: bandName,
    avatar: avatar,
    picture: picture,
    venue: venue,
    start: start,
    venueName: venueName,
    end: end,
  }
  const incrementCartNumber = () => {
    globalCtx.incrementNumber(infos)
  }
  const startDate = new Date(start)
  const endDate = new Date(end)

  const noPicUrl = 'https://picsum.photos/200/300'

  return (
    <Card
      ref={ref}
      sx={{
        width: { xs: 310, md: 345 },
        height: 400,
        margin: { xs: 0, md: 2 },
        my: { xs: 1 },
      }}
    >
      <CardHeader
        avatar={<Avatar alt="Avatar" src={avatar} />}
        title={bandName}
      />
      <CardMedia
        component="img"
        height="140"
        image={picture ? picture : noPicUrl}
        alt="picture"
      />
      <CardContent>
        <Typography gutterBottom variant="h5" component="div">
          <Button target="_blank" href={venue} rel="noopener">
            {' '}
            <LocationOnIcon color="primary" />
            {venueName}{' '}
          </Button>
        </Typography>
        <Typography
          gutterBottom
          variant="body2"
          color="text.secondary"
          component="div"
        >
          {`| Starts: ${startDate.getDate()}.${
            startDate.getMonth() + 1
          }.${startDate.getFullYear()},  ${startDate.getHours()}:${
            startDate.getMinutes() === 0 ? '00' : startDate.getMinutes()
          }`}
        </Typography>
        <Typography variant="body2" color="text.secondary">
          {`| Ends: ${endDate.getDate()}.${
            endDate.getMonth() + 1
          }.${endDate.getFullYear()},  ${endDate.getHours()}:${
            endDate.getMinutes() === 0 ? '00' : endDate.getMinutes()
          }`}
        </Typography>
      </CardContent>
      <CardActions>
        <IconButton aria-label="add" onClick={incrementCartNumber}>
          <AddCircleIcon color="primary" />
        </IconButton>
      </CardActions>
    </Card>
  )
}

export default ItemCard
