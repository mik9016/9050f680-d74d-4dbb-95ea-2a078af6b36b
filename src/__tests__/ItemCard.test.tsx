import React from 'react'
import { act, create } from 'react-test-renderer'
import ItemCard from '../components/ItemCard'

test(`renders ItemCard without crashing`, async () => {
  const img = '../logo.svg'
  const root = create(
    <ItemCard
      id="item['_id']"
      bandName="item['title']"
      avatar={img}
      picture={img}
      start="item['startTime']"
      end="item['endTime']"
      venueName="item['venue']['name']"
      venue="item['venue']['direction']"
      hidePlus={false}
    />
  )

  expect(root.toJSON()).toMatchSnapshot()

  act(() => {
    root.unmount()
  })
})
