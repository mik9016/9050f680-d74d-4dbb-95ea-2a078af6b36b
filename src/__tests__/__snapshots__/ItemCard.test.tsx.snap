// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ItemCard without crashing 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-jehtc8-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
      >
        <img
          alt="Avatar"
          className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="../logo.svg"
        />
      </div>
    </div>
    <div
      className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
      >
        item['title']
      </span>
    </div>
  </div>
  <img
    alt="picture"
    className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
    height="140"
    src="../logo.svg"
  />
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
    >
      <a
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        href="item['venue']['direction']"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        rel="noopener"
        tabIndex={0}
        target="_blank"
      >
         
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
          data-testid="LocationOnIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
          />
        </svg>
        item['venue']['name']
         
      </a>
    </div>
    <div
      className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-186q1z4-MuiTypography-root"
    >
      | Starts: NaN.NaN.NaN,  NaN:NaN
    </div>
    <p
      className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
    >
      | Ends: NaN.NaN.NaN,  NaN:NaN
    </p>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
  >
    <button
      aria-label="add"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
        data-testid="AddCircleIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
    </button>
  </div>
</div>
`;
